File: module4/case_study/src/main/java/com/codegym/model/entity/employee/Employee.java
package com.codegym.model.entity.employee;

import com.codegym.model.entity.contract.Contract;
import com.codegym.model.entity.user.User;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.sql.Date;
import java.util.Set;

@Entity
@Getter
@Setter
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "employee_id")
    private Long id;
    private String employeeName;
    private String employeeBirthday;
    private String employeeIdCard;
    private String employeeSalary;
    private String employeePhone;
    private String employeeEmail;
    private String employeeAddress;
    @ManyToOne
    @JoinColumn(name = "position_id")
    private Position position;
    @ManyToOne
    @JoinColumn(name = "division_id")
    private Division division;
    @ManyToOne
    @JoinColumn(name = "education_degree_id")
    private EducationDegree educationDegree;
    private boolean flag;

    @OneToMany(mappedBy = "employee")
    private Set<Contract> contracts;

    @OneToOne
    @JoinColumn(name = "username", referencedColumnName = "username")
    private User user;

    public Employee() {
    }
}

File: module5/angular-case-study/src/app/model/employee/employee.ts
import {Division} from './division';
import {EducationDegree} from './education-degree';
import {Position} from './position';

export interface Employee {
  id?: number;
  employeeName?: string;
  employeeBirthday?: string;
  employeeIdCard?: string;
  employeeSalary?: number;
  employeePhone?: string;
  employeeEmail?: string;
  employeeAddress?: string;
  division?: Division;
  position?: Position;
  educationDegree?: EducationDegree;
}
